### PR TITLE
more minimap icons

### PR DESCRIPTION
**Pull request by tyeagg**
## About The Pull Request
minimap icons for st, cl ,mo ,cmo, uavs, supply beacons, asrs pads, teleporter pads, sentries, spawners, phero tower and xeno turrets also the sentry, xeno turret flicker when firing, disks and nuke icons flicker when active

image

## Why It's Good For The Game
more information for player

## Changelog
:cl:
imageadd: minimap icons for st, cl ,mo ,cmo, uavs, supply beacons, asrs pads, teleporter pads, sentries, spawners, phero tower and xeno turrets
/:cl:

### CODE REVIEW

File: code/modules/projectiles/sentries.dm [hunk 1]
```diff
@@ -63,8 +65,22 @@
 		camera.c_tag = "[name] ([rand(0, 1000)])"
 
 	GLOB.marine_turrets += src
+	update_minimap_icon()
 	set_on(TRUE)
 
+///Change minimap icon if its firing or not firing
+/obj/machinery/deployable/mounted/sentry/proc/update_minimap_icon()
+	var/marker_flags
+	if(iff_signal == TGMC_LOYALIST_IFF)
```
Comment (TiviPlus) [inline code/modules/projectiles/sentries.dm:74]: remove_marker?

File: code/modules/xenomorph/xeno_structures.dm [hunk 2]
```diff
@@ -877,6 +877,12 @@ TUNNEL
 	var/last_scan_time
 	///light color that gets set in initialize
 	var/light_initial_color = LIGHT_COLOR_GREEN
+	///For minimap icon change if sentry is firing
+	var/firing
+
+///Change minimap icon if its firing or not firing
+/obj/structure/xeno/xeno_turret/proc/update_minimap_icon()
+	SSminimaps.add_marker(src, z, MINIMAP_FLAG_XENO, "xeno_turret[firing ? "_firing" : "_passive"]")
```
Comment (TiviPlus) [inline code/modules/xenomorph/xeno_structures.dm:885]: remove_marker?

File: code/game/objects/machinery/squad_supply/supply_beacon.dm [hunk 3]
```diff
@@ -57,6 +57,16 @@
 	playsound(src, 'sound/machines/twobeep.ogg', 15, 1)
 	H.visible_message("[H] activates [src].",
 	"You activate [src].")
+	var/marker_flags
+	if(H.faction == FACTION_TERRAGOV)
```
Comment (TiviPlus) [inline code/game/objects/machinery/squad_supply/supply_beacon.dm:61]: remove_marker?

File: code/game/objects/machinery/nuclearbomb.dm [hunk 4]
```diff
@@ -324,6 +327,9 @@
 /obj/machinery/nuclearbomb/proc/get_time_left()
 	return timeleft
 
+///Change minimap icon if its on or off
+/obj/machinery/nuclearbomb/proc/update_minimap_icon()
+	SSminimaps.add_marker(src, z, MINIMAP_FLAG_ALL, "nuke[timer_enabled ? "_on" : "_off"]", 'icons/UI_icons/map_blips_large.dmi')
```
Comment (TiviPlus) [inline code/game/objects/machinery/nuclearbomb.dm:332]: remove_marker?

File: code/game/objects/machinery/computer/nuke_disk_generator.dm [hunk 5]
```diff
@@ -152,28 +157,23 @@
 	visible_message(span_notice("[src] beeps as it finishes printing the disc."))
 	reprintable = TRUE
 
+///Change minimap icon if its on or off
+/obj/machinery/computer/nuke_disk_generator/proc/update_minimap_icon()
+	SSminimaps.add_marker(src, z, MINIMAP_FLAG_ALL, "[disk_color]_disk[current_timer ? "_on" : "_off"]", 'icons/UI_icons/map_blips_large.dmi')
```
Comment (TiviPlus) [inline code/game/objects/machinery/computer/nuke_disk_generator.dm:162]: remove_marker?

**Review by TiviPlus — CHANGES_REQUESTED** [review]

File: code/game/objects/machinery/squad_supply/supply_beacon.dm (hunk 3 above)
Comment (tyeagg) [reply code/game/objects/machinery/squad_supply/supply_beacon.dm:61]: not needed on this one remove marker is on deactivate of supply beacon

**Review by Kuro020 — APPROVED** [review]

File: code/__DEFINES/mobs.dm [hunk 6]
```diff
@@ -478,12 +478,12 @@ GLOBAL_LIST_INIT(xenoupgradetiers, list(XENO_UPGRADE_BASETYPE, XENO_UPGRADE_INVA
...
 #define XENO_HEALTH_ALERT_TRIGGER_PERCENT 0.25 //If a xeno is damaged while its current hit points are less than this percent of its maximum, we send out an alert to the hive
 #define XENO_HEALTH_ALERT_TRIGGER_THRESHOLD 50 //If a xeno is damaged while its current hit points are less than this amount, we send out an alert to the hive
-#define XENO_HEALTH_ALERT_COOLDOWN 60 SECONDS //The cooldown on these xeno damage alerts
+#define XENO_HEALTH_ALERT_COOLDOWN 1 MINUTES //The cooldown on these xeno damage alerts
 #define XENO_SILO_HEALTH_ALERT_COOLDOWN 30 SECONDS //The cooldown on these xeno damage alerts
 #define XENO_HEALTH_ALERT_POINTER_DURATION 6 SECONDS //How long the alert directional pointer lasts.
 #define XENO_RALLYING_POINTER_DURATION 15 SECONDS //How long the rally hive pointer lasts
 #define XENO_SILO_DAMAGE_POINTER_DURATION 10 SECONDS //How long the alert directional pointer lasts when silos are damaged
-#define XENO_SILO_DETECTION_COOLDOWN 1 MINUTES
+#define XENO_SILO_DETECTION_COOLDOWN 60 SECONDS
```
Comment (TiviPlus) [inline code/__DEFINES/mobs.dm:486]: What?

File: code/modules/xenomorph/xeno_structures.dm [hunk 7]
```diff
@@ -1240,18 +1268,95 @@ TUNNEL
...
+			. += span_warning("It's slightly damaged, but still seems healthy.")
+		if(80 to 100)
+			. += span_info("It appears in good shape, pulsating healthily.")
+
+
+/obj/structure/xeno/spawner/take_damage(damage_amount, damage_type, damage_flag, sound_effect, attack_dir, armour_penetration)
+	. = ..()
+	spawner_damage_alert()
+
+/obj/structure/xeno/spawner/proc/spawner_damage_alert()
```
Comment (TiviPlus) [inline code/modules/xenomorph/xeno_structures.dm:1312]: autodoc

File: code/modules/xenomorph/xeno_structures.dm [hunk 8]
```diff
@@ -1240,18 +1268,95 @@ TUNNEL
...
+		if(X.hive == associated_hive) //Trigger proxy alert only for hostile xenos
+			return
+
+	warning = TRUE
+	update_minimap_icon()
+	associated_hive.xeno_message("Our [name] has detected a nearby hostile [hostile] at [get_area(hostile)] (X: [hostile.x], Y: [hostile.y]).", "xenoannounce", 5, FALSE, hostile, 'sound/voice/alien_help1.ogg', FALSE, null, /obj/screen/arrow/leader_tracker_arrow)
+	COOLDOWN_START(src, spawner_proxy_alert_cooldown, XENO_SILO_DETECTION_COOLDOWN) //set the cooldown.
+	addtimer(CALLBACK(src, .proc/clear_warning), XENO_SILO_DETECTION_COOLDOWN) //clear warning
+
+/obj/structure/xeno/spawner/proc/clear_warning()
```
Comment (TiviPlus) [inline code/modules/xenomorph/xeno_structures.dm:1348]: autodoc

File: code/modules/xenomorph/xeno_structures.dm [hunk 9]
```diff
@@ -1240,18 +1268,95 @@ TUNNEL
...
+	warning = FALSE
+	update_minimap_icon()
 
 /obj/structure/xeno/spawner/Destroy()
 	GLOB.xeno_spawner -= src
 	return ..()
 
+/obj/structure/xeno/spawner/proc/update_minimap_icon()
```
Comment (TiviPlus) [inline code/modules/xenomorph/xeno_structures.dm:1356]: autodoc

File: code/modules/xenomorph/xeno_structures.dm [hunk 10]
```diff
@@ -832,8 +836,15 @@ TUNNEL
 		if(X.hive == associated_hive) //Trigger proxy alert only for hostile xenos
 			return
 
+	warning = TRUE
+	update_minimap_icon()
 	associated_hive.xeno_message("Our [name] has detected a nearby hostile [hostile] at [get_area(hostile)] (X: [hostile.x], Y: [hostile.y]).", "xenoannounce", 5, FALSE, hostile, 'sound/voice/alien_help1.ogg', FALSE, null, /obj/screen/arrow/leader_tracker_arrow)
 	COOLDOWN_START(src, silo_proxy_alert_cooldown, XENO_SILO_DETECTION_COOLDOWN) //set the cooldown.
+	addtimer(CALLBACK(src, .proc/clear_warning), XENO_SILO_DETECTION_COOLDOWN) //clear warning
+
+/obj/structure/xeno/silo/proc/clear_warning()
```
Comment (TiviPlus) [inline code/modules/xenomorph/xeno_structures.dm:845]: autodoc

File: code/modules/xenomorph/xeno_structures.dm [hunk 11]
```diff
@@ -845,6 +856,10 @@ TUNNEL
 	if(associated_hive)
 		silos += src
 
+/obj/structure/xeno/silo/proc/update_minimap_icon()
```
Comment (TiviPlus) [inline code/modules/xenomorph/xeno_structures.dm:859]: autodoc

**Review by TiviPlus — CHANGES_REQUESTED** [review]
rest lgtm

**Pull request merged**